nomic/dataset.py

```diff
@@ -656,87 +687,86 @@ def _get_atoms(self, ids: List[str]) -> List[Dict]:
-        * **reset_project_if_exists** - If the requested project exists in your organization, will delete it and re-create it.
-        * **project_id** - An alternative way to retrieve a project is by passing the project_id directly. This only works if a project exists.
-        * **reset_project_if_exists** - If the specified project exists in your organization, reset it by deleting all of its data. This means your uploaded data will not be contextualized with existing data.
-        * **add_datums_if_exists** - If specifying an existing project and you want to add data to it, set this to true.
+        * **is_public** - Should this dataset be publicly accessible for viewing (read only). If False, only members of your Nomic organization can view.
+        * **reset_project_if_exists** - If the requested dataset exists in your organization, will delete it and re-create it.
+        * **project_id** - An alternative way to retrieve a dataset is by passing the project_id directly. This only works if a dataset exists.
+        * **reset_project_if_exists** - If the specified dataset exists in your organization, reset it by deleting all of its data. This means your uploaded data will not be contextualized with existing data.
+        * **add_datums_if_exists** - If specifying an existing dataset and you want to add data to it, set this to true.
 
         """
-        assert name is not None or project_id is not None, "You must pass a name or project_id"
+        assert identifier is not None or project_id is not None, "You must pass a dataset identifier"
 
         super().__init__()
 
+        if organization_name is not None:
+            raise DeprecationWarning(f"Passing organization name has been removed in Nomic Python client 3.0. Instead identify your dataset with `organization_name/project_name` (e.g. sterling-cooper/november-ads).")
+
         if project_id is not None:
             self.meta = self._get_project_by_id(project_id)
             return
 
-        if organization_name is None:
-            organization_name = self._get_current_users_main_organization()['nickname']
+        if not self.is_valid_dataset_identifier(identifier=identifier):
+            default_org_slug = self._get_current_users_main_organization()['slug']
+            identifier = default_org_slug+'/'+identifier
 
-        results = self._get_existing_project_by_name(project_name=name, organization_name=organization_name)
+        project = self._get_project_by_slug_identifier(identifier=identifier)
 
-        if 'project_id' in results:  # project already exists
-            organization_name = results['organization_name']
-            project_id = results['project_id']
+        if project:  # dataset already exists
+            project_id = project['id']
             if reset_project_if_exists:  # reset the project
```

Comment:
FWIW I still dislike this flag and think we should get rid of it.

Reply:
yep